spectator/writer/writer_test.go

```diff
@@ -92,3 +92,30 @@ func TestMemoryWriter_Write(t *testing.T) {
 		t.Errorf("expected 10000 lines written to writer but found %d", linesWritten)
 	}
 }
+
+func TestMemoryWriter_Reset(t *testing.T) {
+	w, err := NewWriter("memory", logger.NewDefaultLogger())
+	if err != nil {
+		t.Errorf("failed to create writer: %s", err)
+	}
```

Comment:
You'll need to convert to `MemoryWriter` here using a type conversion:

```suggestion
	w, err := NewWriter("memory", logger.NewDefaultLogger())
	if err != nil {
		t.Errorf("failed to create writer: %s", err)
	}
	mw := w.(*MemoryWriter)
```